### ISSUE TITLE
Which version to use with Java 6

### ISSUE THREAD

**Issue by kells1986** [scalikejdbc/scalikejdbc#737]
I have a large Java 6 project which I'm refactoring to include scalikeJDBC.

So far so good, I really like the library, but when writing some unit tests I've run into an issue.

The dependency I'm using is:

`compile group: 'org.scalikejdbc', name: 'scalikejdbc_2.11', version: '2.4.3'`

I also have this dependency:
 
`compile group: 'commons-dbcp', name: 'commons-dbcp', version: '1.4'`

But when I startup my ConnectionPool during unit testing I get an error:

```
An exception or error caused a run to abort: org/apache/commons/dbcp2/ConnectionFactory : Unsupported major.minor version 51.0 
java.lang.UnsupportedClassVersionError: org/apache/commons/dbcp2/ConnectionFactory : Unsupported major.minor version 51.0
```

When I run the gradle dependency tree I see this:

```
+--- org.scalikejdbc:scalikejdbc-core_2.11:2.4.3
|    |    +--- org.scala-lang:scala-library:2.11.8
|    |    +--- org.scala-lang:scala-reflect:2.11.8 (*)
|    |    +--- org.apache.commons:commons-dbcp2:2.1.1
|    |    |    +--- org.apache.commons:commons-pool2:2.4.2
|    |    |    \--- commons-logging:commons-logging:1.2
```

The library `commons-dbcp2` is a Java 7 based library and so will not work with my project.

Is there another version I should be using that does not include Java 7 dependencies?

**Comment by seratch** [comment]
Although I don’t have a list of connection pool libraries (or older versions) that are compatible with Java 6, there is no barrier if you choose any other connection pool libraries with scalikejdbc.
Old scalikejdbc versions and scala 2.11 are Java 6 compatible. So, only connection pool is the matter.

**Comment by seratch** [comment]
Ah, sorry. I misunderstood. scalikejdbc 2.0 or higher doesn't work with Java 6. You need to use 1.8.x or fork only the connection pool part.

**Comment by kells1986** [comment]
OK, is that available in Maven, as far as I can see only [2.9-2.12](https://repo1.maven.org/maven2/org/scalikejdbc/)  are available.

I followed the readme.md for version 1.8.0 and added these dependencies to a dummy sbt project:

```
libraryDependencies ++= Seq(
  "org.scalikejdbc" %% "scalikejdbc"               % "[1.7,)",
  "org.scalikejdbc" %% "scalikejdbc-interpolation" % "[1.7,)",
  "com.h2database"  %  "h2"                        % "[1.3,)",
  "ch.qos.logback"  %  "logback-classic"           % "[1.0,)"
)
retrieveManaged := true
``` 

My managed_lib directory has `scalikejdbc_2.11-3.0.2.jar` in it.

**Comment by seratch** [comment]
`[1.7,)` doesn't work as you expect. For sbt's lib management, specify `1.8.+` instead.

**Comment by kells1986** [comment]
OK, now I have:
```
libraryDependencies ++= Seq(
  "org.scalikejdbc" %% "scalikejdbc"               % "1.8.+",
  "org.scalikejdbc" %% "scalikejdbc-interpolation" % "1.8.+",
  "com.h2database"  %  "h2"                        % "[1.3,)",
  "ch.qos.logback"  %  "logback-classic"           % "[1.0,)"
)
```

And I get that the dependencies can't be resolved:

```
[warn] ==== public: tried
[warn]   https://repo1.maven.org/maven2/org/scalikejdbc/scalikejdbc-interpolation_2.11/[revision]/scalikejdbc-interpolation_2.11-[revision].pom
[warn]   [3.0.2, 3.0.1, 3.0.0, 3.0.0-RC4, 3.0.0-RC3, 3.0.0-RC2, 3.0.0-RC1, 3.0.0-M4, 3.0.0-M3, 3.0.0-M2, 3.0.0-M1, 2.5.2, 2.5.1, 2.5.0, 2.5.0-RC2, 2.5.0.RC1, 2.4.3, 2.4.2, 2.4.1, 2.4.0, 2.4.0.RC2, 2.4.0.RC1, 2.3.5, 2.3.4, 2.3.3, 2.3.2, 2.3.1, 2.3.0, 2.2.9, 2.2.8, 2.2.7, 2.2.6, 2.2.5, 2.2.4, 2.2.3, 2.2.2, 2.2.1, 2.2.0, 2.1.4, 2.1.3, 2.1.2, 2.1.1, 2.1.0, 2.1.0.RC2, 2.1.0.RC1, 2.0.7, 2.0.6, 2.0.5, 2.0.4, 2.0.3, 2.0.2, 2.0.1, 2.0.0, 2.0.0-RC3, 2.0.0-RC2, 2.0.0-RC1, 2.0.0-beta3, 2.0.0-beta2, 2.0.0-beta1]
[warn] ==== local-preloaded-ivy: tried
[warn]   /Users/jamesk/.sbt/preloaded/org.scalikejdbc/scalikejdbc-interpolation_2.11/[revision]/ivys/ivy.xml
[warn] ==== local-preloaded: tried
[warn]   file:////Users/jamesk/.sbt/preloaded/org/scalikejdbc/scalikejdbc-interpolation_2.11/[revision]/scalikejdbc-interpolation_2.11-[revision].pom
[info] Resolving jline#jline;2.14.3 ...
[warn] 	::::::::::::::::::::::::::::::::::::::::::::::
[warn] 	::          UNRESOLVED DEPENDENCIES         ::
[warn] 	::::::::::::::::::::::::::::::::::::::::::::::
[warn] 	:: org.scalikejdbc#scalikejdbc_2.11;1.8.+: not found
[warn] 	:: org.scalikejdbc#scalikejdbc-interpolation_2.11;1.8.+: not found
[warn] 	::::::::::::::::::::::::::::::::::::::::::::::
[warn] 
[warn] 	Note: Unresolved dependencies path:
[warn] 		org.scalikejdbc:scalikejdbc_2.11:1.8.+ (/Users/jamesk/LearningScala/build.sbt#L13-19)
[warn] 		  +- default:learningscala_2.11:1.0
[warn] 		org.scalikejdbc:scalikejdbc-interpolation_2.11:1.8.+ (/Users/jamesk/LearningScala/build.sbt#L13-19)
[warn] 		  +- default:learningscala_2.11:1.0
[trace] Stack trace suppressed: run last *:update for the full output.
[error] (*:update) sbt.ResolveException: unresolved dependency: org.scalikejdbc#scalikejdbc_2.11;1.8.+: not found
[error] unresolved dependency: org.scalikejdbc#scalikejdbc-interpolation_2.11;1.8.+: not found
[error] Total time: 45 s, completed Aug 3, 2017 11:15:08 AM
```

**Comment by seratch** [comment]
scalikejbdc 1.8 is not available for Scala 2.11. Use Scala 2.10.

**Comment by kells1986** [comment]
ah OK, thanks for your help
